### PR TITLE
[WIP] Add notification bar

### PR DESCRIPTION
**Pull request by biancadanforth**
Notification bar is working!

Some things I did:
* Add try/catch statement to Notification.show method so it doesn't silently fail if 'notice' element is no longer in the DOM. See @osmose's patch (https://github.com/mozilla/pioneer-enrollment-study/commit/b43a1219ea86e7b8aa6f0734a9a4b1e48ea7d83f).
* Add 'studyUtils, 'config' and 'feature' as globals in eslint.
* Replaced local vars 'notice' and 'notificationBox' with class vars 'this.notice' and 'this.notificationBox'
* Replaced "Feature" `Feature.jsm` alias in `bootstrap.js` with "feature" to avoid name collision.
* Added a second button.
* Each button has an empty callback that just print to the console log and send "engaged-prompt" telemetry pings as in the mozilla/pioneer-enrollment-study.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -13,6 +13,11 @@ module.exports = {
           "experimentalObjectRestSpread": true
       }
   },
+  globals: {
```
Comment (gregglind) [inline .eslintrc.js:16]: the better way to do this is to mark it in the file using an eslint comment.  I will show you!

Comment (gregglind) [reply .eslintrc.js:16]: (otherwise, it means that in EVERY file, these will be okay.

File: addon/Config.jsm [hunk 2]
```diff
@@ -11,11 +11,14 @@
 var EXPORTED_SYMBOLS = ["config"];
 
 var config = {
+  // TODO glind: Update notification message as needed
+  notificationMessage: "Please help us improve Firefox and the Web.",
```
Comment (gregglind) [inline addon/Config.jsm:15]: this is going to be part of the settable stuff, and part of the variations array.

File: addon/bootstrap.js [hunk 3]
```diff
@@ -17,12 +15,11 @@ const BASERESOURCE = "perception-study";
 
 XPCOMUtils.defineLazyModuleGetter(this, "studyUtils",
   `resource://${BASERESOURCE}/StudyUtils.jsm`);
-XPCOMUtils.defineLazyModuleGetter(this, "Feature",
+XPCOMUtils.defineLazyModuleGetter(this, "feature",
```
Comment (gregglind) [inline addon/bootstrap.js:18]: I am not sure what exactly we are going to export here, the class or an instance :)

File: addon/bootstrap.js [hunk 4]
```diff
@@ -98,9 +93,8 @@ async function startup(addonData, reason) {
 
     });
   }
-
-  console.log(new Feature().startup());
-
+  
+  feature.startup(variation.name);
```
Comment (gregglind) [inline addon/bootstrap.js:97]: Same, classname vs instance :)  Sometimes this stuff goes back and forth :)

File: addon/lib/Feature.jsm [hunk 5]
```diff
@@ -8,53 +8,108 @@ Cu.import("resource://gre/modules/XPCOMUtils.jsm");
...
+XPCOMUtils.defineLazyModuleGetter(this, "studyUtils",
+  `resource://${BASERESOURCE}/StudyUtils.jsm`);
 
 class Feature {
-  startup(config) {
-    console.log(`Starting the thing with ${config}`);
-    return "started";
+  constructor() {
+    this.notificationBar = new Notification();
+    this.TREATMENTS = {
```
Comment (gregglind) [inline addon/lib/Feature.jsm:25]: so, this isn't aactually what we want, structurally.  we just want one treatment, but I am not sure here.

File: addon/lib/Feature.jsm [hunk 6]
```diff
@@ -8,53 +8,108 @@ Cu.import("resource://gre/modules/XPCOMUtils.jsm");
...
 
+  getMostRecentBrowserWindow() {
+    return RecentWindow.getMostRecentBrowserWindow({
+      private: false,
+      allowPopups: false,
+    });
   }
 }
 
 class Notification {
```
Comment (gregglind) [inline addon/lib/Feature.jsm:74]: Notification class:  looks good.

File: addon/lib/Feature.jsm [hunk 7]
```diff
@@ -8,53 +8,108 @@ Cu.import("resource://gre/modules/XPCOMUtils.jsm");
...
     this.notificationBox = null;
   }
-  show(doc, onClickButtonCallback) {
-    let { notice, notificationBox} = this
+  show(doc, onClickButtonAcceptCallback, onClickButtonRejectCallback) {
 
     // only one at a time is allowed
-    if (notice && notificationBox) {
-      notificationBox.removeNotification(notice);
+    if (this.notice && this.notificationBox) {
```
Comment (gregglind) [inline addon/lib/Feature.jsm:84]: I think changing these all back to "this" accessors is worse, and harder to read everywhere

**Review by gregglind — COMMENTED** [review]
This touches too many things at once, and is hard to accept as a full patch.  I am going to take the pieces that work and bring them in.

File: addon/lib/Feature.jsm (hunk 5 above)
Comment (biancadanforth) [reply addon/lib/Feature.jsm:25]: Well my thinking was TREATMENTS were associated with the feature, so should be in the Feature class. There would only be one TREATMENTS assuming we only create one instance of the Feature. This was just an attempt to minimize the number of global variables. Feel free to move around!

**Comment by gregglind** [comment]
All changes from this patch are superceded or incoporated in #3, closing.

**Pull request closed without merging**